src/commands/deployWorkspaceProject/getDefaultValues/throwIfTreeItemAndWorkspaceSettingsConflict.ts

```diff
@@ -0,0 +1,55 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import type { ContainerApp, ContainerAppsAPIClient } from "@azure/arm-appcontainers";
+import { ISubscriptionActionContext, nonNullValue } from "@microsoft/vscode-azext-utils";
+import { ext } from "../../../extensionVariables";
+import { ContainerAppItem } from "../../../tree/ContainerAppItem";
+import { ManagedEnvironmentItem } from "../../../tree/ManagedEnvironmentItem";
+import { createContainerAppsAPIClient } from "../../../utils/azureClients";
+import { localize } from "../../../utils/localize";
+import type { DeployWorkspaceProjectSettings } from "../deployWorkspaceProjectSettings";
+
+/**
+ * @throws Throws an error if there is a deployment target mismatch between the tree item and workspace settings provided
+ */
+export async function throwIfTreeItemAndWorkspaceSettingsConflict(
```

Comment:
New code

Reply:
I mentioned it in a comment above, but I think we should override rather than throw an error when there are mis-matched settings. I think we can prompt the user to let them know and maybe ask if they want to change their default settings as well.